src/components/ColorRampDropdown.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { ReactElement, useContext, useEffect, useMemo, useState } from "react";
+import styles from "./ColorRampDropdown.module.css";
+import { DEFAULT_COLOR_RAMPS } from "../constants/color_ramps";
+import { Button } from "antd";
+import { AppThemeContext } from "./AppStyle";
+import AccessibleTooltip from "./AccessibleTooltip";
+
+type ColorRampSelectorProps = {
+  selected: string;
+  onChange: (colorRampKey: string) => void;
+  colorRamps?: typeof DEFAULT_COLOR_RAMPS;
+  disabled?: boolean;
+};
+
+const defaultProps: Partial<ColorRampSelectorProps> = {
+  colorRamps: DEFAULT_COLOR_RAMPS,
+  disabled: false,
+};
+
+/**
+ * A dropdown selector for color ramp gradients.
+ */
+const ColorRampSelector: React.FC<ColorRampSelectorProps> = (propsInput): ReactElement => {
+  const props = { ...defaultProps, ...propsInput } as Required<ColorRampSelectorProps>;
+  const theme = useContext(AppThemeContext);
+
+  // Force the dropdown to stay open when clicked for accessibility. Close it again
+  // when focus is lost.
+  const [forceOpen, setForceOpen] = useState(false);
+  const componentContainerRef = React.useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (!forceOpen) {
+      return;
+    }
+    const doesContainTarget = (target: EventTarget | null): boolean => {
+      return (
+        (target instanceof Element &&
+          componentContainerRef.current &&
+          componentContainerRef.current.contains(target)) ||
+        false
+      );
+    };
+    // Handle focus loss for tab navigation
+    const handleFocusLoss = (event: FocusEvent): void => {
+      if (!doesContainTarget(event.relatedTarget)) {
+        setForceOpen(false);
+      }
+    };
+
+    componentContainerRef.current?.addEventListener("focusout", handleFocusLoss);
+    return () => {
+      componentContainerRef.current?.removeEventListener("focusout", handleFocusLoss);
+    };
+  }, [forceOpen]);
```

Comment:
This is most of the new content in `ColorRampDropdown`, plus some className manipulation below.

Reply:
(sorry that the diff is ugly, it didn't figure out that I renamed this file)

Reply:
oh wait nvm GitHub figured it out!